src/main/java/com/google/sps/servlets/EventsServlet.java

```diff
@@ -0,0 +1,199 @@
+        Date endTime = (Date) entity.getProperty(END_TIME);
+        String category = (String) entity.getProperty(CATEGORY);
+
+        DateTime startDateTime = new DateTime(startTime);
+        EventDateTime start = new EventDateTime()
+            .setDateTime(startDateTime)
+            .setTimeZone(UTC_TIMEZONE);
+        DateTime endDateTime = new DateTime(endTime);
+        EventDateTime end = new EventDateTime()
+            .setDateTime(endDateTime)
+            .setTimeZone(UTC_TIMEZONE);
+        Event event = new Event()
+            .setSummary(summary)
+            .setLocation(location)
+            .setDescription(description)
+            .setStart(start)
+            .setEnd(end);
+
+        ExtendedProperties ep = new ExtendedProperties();
+        ep.set(CATEGORY, category);
+        event.setExtendedProperties(ep);
+    
+        events.add(event);
+    }
+
+    response.setContentType("application/json; charset=UTF-8");
+    response.setCharacterEncoding("UTF-8");
+
+    String json = convertToJson(events);
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String eventSummary = request.getParameter(SUMMARY);
+      String eventDescription = request.getParameter(DESCRIPTION);
+      String eventLocation = request.getParameter(LOCATION);
+      String eventDateString = request.getParameter(DATE);
+      String eventStartTimeString = request.getParameter(START_TIME);
+      String eventEndTimeString = request.getParameter(END_TIME);
+      String timezoneOffset = request.getParameter(USER_TIMEZONE);
+      String category = request.getParameter(CATEGORY);
+
+      if (!CATEGORIES.contains(category)) category = "other";
+  
+      Date eventStartDateTime = getEventDateTime(eventDateString, eventStartTimeString, timezoneOffset);
+      Date eventEndDateTime = getEventDateTime(eventDateString, eventEndTimeString, timezoneOffset);
+      if (eventStartDateTime == null || eventEndDateTime == null) return;
+      long timestamp = System.currentTimeMillis();
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+      Entity eventEntity = new Entity(EVENT);
+      eventEntity.setProperty(SUMMARY, eventSummary);
+      eventEntity.setProperty(TIMESTAMP, timestamp);
+      eventEntity.setProperty(LOCATION, eventLocation);
+      eventEntity.setProperty(DESCRIPTION, eventDescription);
+      eventEntity.setProperty(START_TIME, eventStartDateTime);
+      eventEntity.setProperty(END_TIME, eventEndDateTime);
+      eventEntity.setProperty(CATEGORY, category);
+
+      datastore.put(eventEntity);
+
+      response.sendRedirect("/feed.html");
+  }
+
+  private String convertToJson(List<Event> events) {
+      Gson gson = new Gson();
+      String json = gson.toJson(events);
+      return json;
+  }
+
+  public Date getEventDateTime(String eventDate, String eventTime, String timezoneOffset) {
```

Comment:
As this method does some non-trivial computation that isn't entirely captured by the method's name, add a comment describing what getEventDateTime does and the form each input parameter is expected to take.